custom_components/gtfs_rt/PublicTransportData.py

```diff
@@ -0,0 +1,382 @@
+        debug_dataframe(trip_update_df, "Trip Update")
+
+        # create vehicle info dataframe
+        v_feed_entities = get_gtfs_feed_entities(
+            url=self._vehicle_position_url,
+            headers=self._headers,
+            label="vehicle positions",
+        )
+        vehicle_info_df = gtfs_data_to_df(v_feed_entities, label="vehicle")
+        debug_dataframe(vehicle_info_df, "Vehicle Info")
+
+        # add vehicle to trip update info TODO: fix merging- some missing?
+        trip_update_df = pd.merge(
+            left=trip_update_df, right=vehicle_info_df, how="left"
+        )
+        debug_dataframe(trip_update_df, "Merged Trip & Vehicle Info")
+
+        # get static timetable data
+        timetable_df = StaticMasterGTFSInfo(
+            url=self.static_gtfs_url, CachedData=self.CachedGTFSData
+        ).departure_info
+        debug_dataframe(timetable_df, "Static Timetable Info")
+
+        # merge live and static data
+        trip_update_df = pd.merge(
+            trip_update_df,
+            timetable_df,
+            how="outer",
+            on="trip_id",
+            indicator="source",
+        )
+
+        # convert categorical columns to strings
+        trip_update_df[
+            [
+                "stop_id_x",
+                "stop_id_y",
+                "route_id_x",
+                "route_id_y",
+                "direction_id_x",
+                "direction_id_y",
+            ]
+        ] = trip_update_df[
+            [
+                "stop_id_x",
+                "stop_id_y",
+                "route_id_x",
+                "route_id_y",
+                "direction_id_x",
+                "direction_id_y",
+            ]
+        ].astype(
+            str
+        )
+        # choose non-NA values for stop_id, route_id, and direction_id
+        # combine stop_id, route_id, and direction_id columns
+        trip_update_df["stop_id"] = trip_update_df["stop_id_x"].combine_first(
+            trip_update_df["stop_id_y"]
+        )
+        trip_update_df["route_id"] = trip_update_df[
+            "route_id_x"
+        ].combine_first(trip_update_df["route_id_y"])
+        trip_update_df["direction_id"] = trip_update_df[
+            "direction_id_x"
+        ].combine_first(trip_update_df["direction_id_y"])
+
+        # drop intermediate columns
+        trip_update_df = trip_update_df.drop(
+            columns=[
+                "stop_id_x",
+                "stop_id_y",
+                "route_id_x",
+                "route_id_y",
+                "direction_id_x",
+                "direction_id_y",
+            ]
+        )
+        debug_dataframe(trip_update_df, "Merged live and static info")
+
+        # need to split route_ids if there's a delimiter
+        if self._route_delimiter is not None:
+            trip_update_df["route_id", "route_id_split"] = trip_update_df[
+                "route_id"
+            ].str.split(self._route_delimiter, expand=True)
```

Comment:
Need to add an indicator for if this is trip got a "Live" update as opposed to just being based on the schedule.

Reply:
Done - "real_time_update" column created.